iphone/Maps/Classes/Components/MWMTableViewController.m

```diff
@@ -31,6 +31,25 @@ - (void)viewDidLoad
          forCellReuseIdentifier:[MWMTableViewSubtitleCell className]];
 }
 
+// Fix table section header font color for all tables, including Setting and Route Options.
+- (void)tableView:(UITableView *)tableView willDisplayHeaderView:(UIView *)view forSection:(NSInteger)section {
+  if ([view isKindOfClass: [UITableViewHeaderFooterView class]]) {
+    UITableViewHeaderFooterView * header = (UITableViewHeaderFooterView *)view;
+    UIColor * sectionHeaderColor = [UIColor blackSecondaryText];
+    @try {
```

Comment:
No, @available is not enough here. I'm using a private Apple API to avoid any overhead. It means that these properties may be _absent_ when they are accessed by valueForKey, and throw an exception. In the worst case (e.g. a new iOS version changes internals) the exception will fire and nothing happens. The Light mode will work the same and the text will be readable. In the Dark mode, the text won't be really readable, but the app won't crash.

I know that this approach has its cons. Are there any cleaner ways to override the dark theme color in such a case?

Reply:
Got it. 
It seems the only way to do it "right" would be to use fully custom header view, but it requires much more efforts rather than simple "hack" like you've implemented here

In general `@try/@catch` in Objective C is really rarely used and people usually try to avoid it, you can try to use smth like [`-[NSObject respondsToSelector:]`](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418583-respondstoselector) instead(it dynamically checks if certain object has method), but tbh I don't see any real problems with this try/catch - after all current solution just works and I have no other arguments why implementation without try/catch would be better rather than "idiomacity" :)

Reply:
Can you please try it with respondsToSelector and test it on iOS 12, 13, and 14+ simulators? I think I had some issues with that approach and it didn't work.

Reply:
What is the easiest way to build Organic Maps in Xcode? I guess I can just open `xcode/omim.xcworkspace` and it should work?

I'll be honest: I do not promise to do it soon as I am a bit limited in time(and want to focus on https://github.com/organicmaps/organicmaps/pull/4022/files in the first order as I already started it)

Reply:
Yes. Don't forget to run ./configure.sh

Reply:
Hm, looked into it for 5 min:
1. Indeed it didn't work for me with `respondsToSelector`(not sure why, but didn't dig deep)
2. I quickly tried such variant on iOS 16:
```
// Fix table section header font color for all tables, including Setting and Route Options.
- (void)tableView:(UITableView *)tableView willDisplayHeaderView:(UIView *)view forSection:(NSInteger)section {
  if ([view isKindOfClass: [UITableViewHeaderFooterView class]]) {
    UITableViewHeaderFooterView * header = (UITableViewHeaderFooterView *)view;
    UIColor * sectionHeaderColor = [UIColor blackSecondaryText];
    header.textLabel.textColor = sectionHeaderColor;
  }
}
```
It seems to work, but the only problem with it is that [`textLabel` property is deprecated](https://developer.apple.com/documentation/uikit/uitableviewheaderfooterview/1624912-textlabel?language=objc) and suggested alternative is non-trivial to use(at least at glance). But at least it doesn't rely on private APIs...

Reply:
Btw current solution has small bug when we change day/night mode manually: 

0. We are in day mode
1. Go to settings 
2. Go to "Night mode"
3. Enable night mode
4. Go back to Settings

Actual: headers are still in day mode
Expected: headers should change to night mode

Not sure how it works for the rest of UI, but I fixed it with:
```
- (void)viewWillAppear:(BOOL)animated
{
  [super viewWillAppear:animated];
  [self.tableView reloadData];
}
```
in `MWMTableViewController`(but I guess there is a better way as the rest of UI is redrawn automatically)

Reply:
> I quickly tried such variant on iOS 16:

Please check it with iOS 12 and 13 too. Deprecation is not an issue until our min supported iOS version is 12 or 13.

Reloading the whole table is an overkill. There are a few cleaner solutions here, but they require to init and keep pointers to these custom headers: https://stackoverflow.com/questions/23501952/refresh-only-the-custom-header-views-in-a-uitableview

Reply:
@biodranik 
See:
<img width="1533" alt="Screenshot 2022-12-05 at 17 51 48" src="https://user-images.githubusercontent.com/266271/205695191-986d7e8d-4cbb-4477-933b-2f89a011f156.png">

Final version of code(together with fix of bug I found):
```
diff --git a/iphone/Maps/Classes/Components/MWMTableViewController.m b/iphone/Maps/Classes/Components/MWMTableViewController.m
index e53286a0f0..3dd7c90456 100644
--- a/iphone/Maps/Classes/Components/MWMTableViewController.m
+++ b/iphone/Maps/Classes/Components/MWMTableViewController.m
@@ -19,6 +19,24 @@ - (BOOL)prefersStatusBarHidden
   return NO;
 }
 
+- (void)viewWillAppear:(BOOL)animated
+{
+  [super viewWillAppear:animated];
+  
+  for (NSInteger i = 0; i < [self numberOfSectionsInTableView:self.tableView]; i++) {
+    [self updateHeaderView:[self.tableView headerViewForSection:i]];
+  }
+}
+
+// Fix table section header font color for all tables, including Setting and Route Options.
+- (void)updateHeaderView:(UIView*)headerView {
+  if ([headerView isKindOfClass: [UITableViewHeaderFooterView class]]) {
+    UITableViewHeaderFooterView * header = (UITableViewHeaderFooterView *)headerView;
+    UIColor* sectionHeaderColor = [UIColor blackSecondaryText];
+    header.textLabel.textColor = sectionHeaderColor;
+  }
+}
+
 - (void)viewDidLoad
 {
   [super viewDidLoad];
@@ -31,6 +49,11 @@ - (void)viewDidLoad
          forCellReuseIdentifier:[MWMTableViewSubtitleCell className]];
 }
 
+- (void)tableView:(UITableView *)tableView willDisplayHeaderView:(UIView *)view forSection:(NSInteger)section {
+  [self updateHeaderView:view];
+}
+
+
 #pragma mark - Properties
 
 - (BOOL)hasNavigationBar
```

Reply:
Cool, your fix is way cleaner 👍 Can you please create a PR, and mention that it closes this one in the description?